src/pages/iou/request/step/IOURequestStepParticipants.js

```diff
@@ -0,0 +1,103 @@
+import MoneyRequestParticipantsSelector from '@pages/iou/request/MoneyTemporaryForRefactorRequestParticipantsSelector';
+import * as IOU from '@userActions/IOU';
+import CONST from '@src/CONST';
+import ROUTES from '@src/ROUTES';
+import IOURequestStepRoutePropTypes from './IOURequestStepRoutePropTypes';
+import StepScreenWrapper from './StepScreenWrapper';
+import withFullTransactionOrNotFound from './withFullTransactionOrNotFound';
+import withWritableReportOrNotFound from './withWritableReportOrNotFound';
+
+const propTypes = {
+    /** Navigation route context info provided by react navigation */
+    route: IOURequestStepRoutePropTypes.isRequired,
+
+    /* Onyx Props */
+    /** The transaction object being modified in Onyx */
+    transaction: transactionPropTypes,
+};
+
+const defaultProps = {
+    transaction: {},
+};
+
+function IOURequestStepParticipants({
+    route: {
+        params: {iouType, reportID, transactionID},
+    },
+    transaction,
+    transaction: {participants = []},
+}) {
+    const {translate} = useLocalize();
+    const optionsSelectorRef = useRef();
+    const selectedReportID = useRef(reportID);
+    const numberOfParticipants = useRef(participants.length);
+    const iouRequestType = TransactionUtils.getRequestType(transaction);
+    const headerTitle = translate(TransactionUtils.getHeaderTitleTranslationKey(transaction));
+    const receiptFilename = lodashGet(transaction, 'filename');
+    const receiptPath = lodashGet(transaction, 'receipt.source');
+
+    // When the component mounts, if there is a receipt, see if the image can be read from the disk. If not, redirect the user to the starting step of the flow.
+    // This is because until the request is saved, the receipt file is only stored in the browsers memory as a blob:// and if the browser is refreshed, then
+    // the image ceases to exist. The best way for the user to recover from this is to start over from the start of the request process.
+    useEffect(() => {
+        IOUUtils.navigateToStartStepIfScanFileCannotBeRead(receiptFilename, receiptPath, () => {}, iouRequestType, iouType, transactionID, reportID);
+    }, [receiptPath, receiptFilename, iouRequestType, iouType, transactionID, reportID]);
+
+    const addParticipant = useCallback(
+        (val) => {
+            IOU.setMoneyRequestParticipants_temporaryForRefactor(transactionID, val);
+            numberOfParticipants.current = val.length;
+
+            // When multiple participants are selected, the reportID is generated at the end of the confirmation step.
+            if (val.length !== 1) {
+                return;
+            }
+
+            // When a participant is selected, the reportID needs to be saved because that's the reportID that will be used in the confirmation step.
+            selectedReportID.current = lodashGet(val, '[0].reportID', reportID);
+        },
+        [reportID, transactionID],
+    );
+
+    const goToNextStep = useCallback(() => {
```

Comment:
We missed that the participants needed to be reset here (as was done in `MoneyRequestParticipantsPage` previously), this caused https://github.com/Expensify/App/issues/33067 where tags and categories were not resetting when switching workspace for a request. More details [here](https://github.com/Expensify/App/issues/33067#issuecomment-1855974499).